### PR TITLE
Accessibility - Added missing image ALT tags, search form title attribute and corrected headings order on homepage

### LINKED ISSUE #3291: Accessibility - Missing image ALT tags and wrong headings order
Branch: develop
Expected Behavior: "All non-text content that is presented to the user has a text alternative that serves the equivalent purpose." (WCAG 2.0 Guideline 1.1.1.)

WAVE checker is also complaining about incorrectly ordered headings (WCAG 2.0 Guideline 2.4.6)

Current Behavior: Current ka-lite server **homepage** has a CC logo in the footer and the linked "Learn" image without the appropriate ALT tags. 

"A free world-class education for anyone anywhere." title should be H1 instead of H2

(image is from the master branch, so slightly different from develop)
homepage-wave-errors

### PR DESCRIPTION
**Pull request by radinamatic**
issue #3291
homepage-alt-h1-corrected

Fixes #3291

### CODE REVIEW

File: kalite/distributed/templates/distributed/base.html [hunk 1]
```diff
@@ -170,8 +170,8 @@
                                   <li id="search_input">
                                       <form class="navbar-form " action="{% url 'search' %}" method="get"  role="search">
                                           {% comment %}translators: this will appear in the navigation bar. please try to keep it as short as possible.{% endcomment %}
-                                          <input type="text" name="query" id="search_box" class="form-control pull-left" placeholder="{% trans 'Topic, video, exercise...' %}" />
-                                          <button class="btn btn-default" id="search_btn" type="submit"><i class="glyphicon glyphicon-search"></i></button>
+                                          <input type="text" name="query" id="search_box" class="form-control pull-left" title="Type search term here" placeholder="{% trans 'Topic, video, exercise...' %}" />
```
Comment (rtibbles) [inline kalite/distributed/templates/distributed/base.html]: Please wrap these strings in Django translation tags. {% trans '' %}

File: kalite/distributed/templates/distributed/base.html [hunk 2]
```diff
@@ -182,8 +182,8 @@
                               <form class="navbar-form" action="{% url 'search' %}" method="get" id="search-box" role="search">
                                 {% comment %}translators: this will appear in the navigation bar. please try to keep it as short as possible.{% endcomment %}
                                 <div class="input-group">
-                                  <input type="text" name="query" id="search" class="form-control" placeholder="{% trans 'Topic, video, exercise...' %}" />
-                                  <button id="search-button" class="btn btn-default" type="submit"><i class="glyphicon glyphicon-search"></i></button>
+                                  <input type="text" name="query" id="search" class="form-control" title="Type search term here" placeholder="{% trans 'Topic, video, exercise...' %}" />
+                                  <button id="search-button" class="btn btn-default" type="submit" value="Search"><i class="glyphicon glyphicon-search"></i></button>
```
Comment (rtibbles) [inline kalite/distributed/templates/distributed/base.html]: I don't know enough about how screen readers work to understand the purpose of the 'value' attribute here - can you explain?

File: kalite/distributed/templates/distributed/homepage.html [hunk 3]
```diff
@@ -62,14 +62,14 @@ <h2 class="playlist-list-header"><%= title %></h2>
...
+            <h1 class="main-headline hidden-xs">{{ channel_data.head_line }}</h1>
         </div>
 
         <div class="row">
             <div class="col-sm-6 col-sm-push-3 col-xs-8 col-xs-push-2 vertical-shadow suggested-action">
                 <a href="{% url 'learn' %}">
-                    <h3 class="suggested-action-title">{% trans "Learn!" %}</h3>
-                    <img id="suggested-action-image-link-homepage" src="/data/{{ channel }}/images/learning-image.png">
+                    <h2 class="suggested-action-title">{% trans "Learn!" %}</h2>
+                    <img id="suggested-action-image-link-homepage" src="/data/{{ channel }}/images/learning-image.png"  alt="Click to start learning.">
```
Comment (rtibbles) [inline kalite/distributed/templates/distributed/homepage.html]: Translation tags for string here too.

File: kalite/distributed/templates/distributed/homepage.html [hunk 4]
```diff
@@ -62,14 +62,14 @@ <h2 class="playlist-list-header"><%= title %></h2>
     <div class="container">
 
         <div class="row">
-            <h2 class="main-headline hidden-xs">{{ channel_data.head_line }}</h2>
```
Comment (rtibbles) [inline kalite/distributed/templates/distributed/homepage.html]: I don't think the h1 tag is appropriate here, the default content for this is "A free world-class education for anyone anywhere." which is not a page heading (h1) but rather a subheading - <p> may be more appropriate.

File: kalite/distributed/templates/distributed/homepage.html [hunk 5]
```diff
@@ -62,14 +62,14 @@ <h2 class="playlist-list-header"><%= title %></h2>
...
+            <h1 class="main-headline hidden-xs">{{ channel_data.head_line }}</h1>
         </div>
 
         <div class="row">
             <div class="col-sm-6 col-sm-push-3 col-xs-8 col-xs-push-2 vertical-shadow suggested-action">
                 <a href="{% url 'learn' %}">
-                    <h3 class="suggested-action-title">{% trans "Learn!" %}</h3>
-                    <img id="suggested-action-image-link-homepage" src="/data/{{ channel }}/images/learning-image.png">
+                    <h2 class="suggested-action-title">{% trans "Learn!" %}</h2>
```
Comment (rtibbles) [inline kalite/distributed/templates/distributed/homepage.html]: Again, heading may not be the semantically appropriate tag here.

File: kalite/distributed/templates/distributed/partials/_footer.html [hunk 6]
```diff
@@ -17,7 +17,7 @@
                         {{ channel_data.footer_text }}
                     </p>
                     <p class="footer-text">
-                        <img id="creative-commons-icon" src="{% static 'images/distributed/' True %}{{ channel_data.channel_license }}.png">
+                        <img id="creative-commons-icon" src="{% static 'images/distributed/' True %}{{ channel_data.channel_license }}.png" alt="Creative Commons logo">
```
Comment (rtibbles) [inline kalite/distributed/templates/distributed/partials/_footer.html]: This is *not* the Creative Commons logo. Rather it is a specific license, in the case of our default Khan Academy content it is "Creative Commons Attribution-NonCommercial-ShareAlike 3.0 United States License" see Creative Commons for details.

This should also be wrapped in translation tags.

**Comment by rtibbles** [comment]
Changes strings - targeting to bugfixes.

File: kalite/distributed/templates/distributed/base.html (hunk 1 above)
Comment (radinamatic) [inline kalite/distributed/templates/distributed/base.html]: Done & done!

File: kalite/distributed/templates/distributed/base.html (hunk 2 above)
Comment (radinamatic) [inline kalite/distributed/templates/distributed/base.html]: @rtibbles, sorry for my late reply, messy days at uni.

Strictly speaking, 'value' attribute may not be the requirement here as it would be for input field with type="submit". However, screen readers tend to have similar issues as browsers: even though specification says one thing, they may render speech with slight differences, and in the absence of descriptive text labels they rely on "surroundings" to infer (aka guess ;) the meaning and purpose of control elements. 

Take for example the following block of code:

```html
<input title="Type your search term here" placeholder="Topic, video, exercise..." type="text">
<button type="submit" value="Search"><i class="glyphicon glyphicon-search"></i></button>
```
It gives these speech outputs in 3 different screen readers:

**JAWS**

Topic video exercise dot dot dot 
edit Type your search term here 
type and text
button to activate press space bar

**NVDA**

Topic video exercise
edit Type your search term here
blank
button

**ZoomText**

Topic video exercise
edit box
button

Similarly, in our full code with ARIA roles implemented, JAWS reads "search landmark region 
Topic video exercise dot dot dot" when input field is in focus, while NVDA reads "search landmark button" when the search button is in focus, ZoomText doesn't seem to see neither ARIA roles nor title attribute, and other screen readers may render something completely different.

To sum up: even though not strictly required, value="Search" in the button element may give additional inferential cues to some screen readers in the absence of descriptive text label.

Hope this helps... :)

**Comment by rtibbles** [comment]
Hi @radinamatic looking at the files changed now, it doesn't seem that any of your accessibility changes have actually made it through the merge conflict resolution wringer.

Could you open a new PR with your changes committed on top of latest develop to make sure we can get them in?
